Content.Server/_NF/Cargo/Systems/CargoSystem.TradeCrates.cs

```diff
@@ -0,0 +1,110 @@
+using System.Threading;
+using Content.Server._NF.Trade;
+using Content.Shared._NF.Trade;
+using Content.Shared.Examine;
+using Content.Shared.GameTicking;
+using Robust.Shared.GameObjects;
+using Timer = Robust.Shared.Timing.Timer;
+
+namespace Content.Server.Cargo.Systems; // Needs to collide with base namespace
+
+public sealed partial class CargoSystem
+{
+    private List<EntityUid> _destinations = new();
+
+    private void InitializeTradeCrates()
+    {
+        SubscribeLocalEvent<TradeCrateComponent, PriceCalculationEvent>(OnTradeCrateGetPriceEvent);
+        SubscribeLocalEvent<TradeCrateComponent, ComponentInit>(OnTradeCrateInit);
+        SubscribeLocalEvent<TradeCrateComponent, ComponentRemove>(OnTradeCrateRemove);
+        SubscribeLocalEvent<TradeCrateComponent, ExaminedEvent>(OnTradeCrateExamined);
+
+        SubscribeLocalEvent<TradeCrateDestinationComponent, ComponentInit>(OnDestinationInit);
+        SubscribeLocalEvent<TradeCrateDestinationComponent, ComponentRemove>(OnDestinationRemove);
+    }
+
+    private void OnTradeCrateGetPriceEvent(EntityUid uid, TradeCrateComponent component, ref PriceCalculationEvent ev)
```

Comment:
entity<T>

Reply:
What does using Entity<T> over Entity, T give you?

Reply:
Entity<T> is the preferred signature for modern event handlers and Public API methods, every other example is purely a hangover and shouldn't be merged

You can check it out in the [docs](https://docs.spacestation14.com/en/robust-toolbox/ecs.html?highlight=by-ref#events-subscriptions-and-methods), although don't discard event args like it says you can, because you actually can't and it will just [create a new variable ](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards#a-standalone-discard)with the `_` signature.

Reply:
Was looking for a bit more of an answer than "we don't do that anymore."

Reply:
it exists so you can easily pair up the entityuid and components, and its how everything should be handled now, theres probably not much else to it but idk ask wizden

![image](https://github.com/user-attachments/assets/db5239ee-4936-41fb-810a-5fdef30c4307)